modules/ezcontent_node/modules/ezcontent_smart_article/ezcontent_smart_article.module

```diff
@@ -144,9 +145,99 @@ function ezcontent_smart_article_form_node_form_alter(&$form, FormStateInterface
             $form[$term_reference_field]['widget']['smart_tags_submit']['#attributes'] = empty($summary_config->get('smart_tags_api_url')) ? ['disabled' => 'disabled'] : [];
           }
         }
+
+        // Custom submit handler for Podcast.
+        foreach ($entity->getFieldDefinitions() as $field_definition) {
+          if ($field_definition->getType() === 'ezcontent_smart_podcast') {
+            // Submit handler for podcast.
+            $form['actions']['submit']['#submit'][] = '_custom_submit_handler';
+            break;
+          }
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Custom submit handler for podcast.
+ *
+ * @param array $form
+ *   The form object.
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   The form state object.
+ */
+function _custom_submit_handler(array &$form, FormStateInterface $form_state) {
+  $text = '';
+  $entity = $form_state->getFormObject()->getEntity();
+  foreach ($entity->getFieldDefinitions() as $field_definition) {
+    if ($field_definition->getType() === 'ezcontent_smart_podcast') {
+      $field_name = $field_definition->getName();
+      $text_to_speech_fields = $field_definition->getSettings()['text_to_speech_fields'];
+    }
+  }
+  if (isset($field_name) && isset($text_to_speech_fields)) {
+    if ($entity->$field_name->convert_text_to_speech) {
+      $data['entity_type_id'] = $entity->getEntityTypeId();
+      $data['entity_id'] = $entity->id();
+      $data['field_name'] = $field_name;
+      $text_to_speech_content_fields = $text_to_speech_fields['content'];
+      $text_to_speech_reference_fields = $text_to_speech_fields['paragraphs'];
+      foreach ($text_to_speech_content_fields as $text_to_speech_field) {
+        if (in_array($text_to_speech_field, array_keys($text_to_speech_reference_fields))) {
```

Comment:
Can you try this `isset($text_to_speech_reference_fields[$text_to_speech_field]) ....`